Source/WebKit/UIProcess/ProcessThrottler.cpp

```diff
@@ -257,6 +269,18 @@ bool ProcessThrottler::isValidForegroundActivity(const ProcessThrottler::Activit
     return std::get<UniqueRef<ProcessThrottler::ForegroundActivity>>(activity)->isValid();
 }
 
+void ProcessThrottler::setAllowActivity(bool allow)
+{
+    if (m_allowActivity == allow)
+        return;
+
+    PROCESSTHROTTLER_RELEASE_LOG("setAllowActivity %d", allow);
+    if (!allow)
+        invalidateAllActivities();
+
+    m_allowActivity = allow;
```

Comment:
I think we'd better set this flag before calling `invalidateAllActivities()`. We wouldn't want code getting called from `invalidateAllActivities()` to retake new activities if allow is false.